src/components/ChatMessages.tsx

```diff
@@ -159,13 +159,13 @@ export default class ChatMessages extends React.Component<ChatMessagesProps> {
   renderMessageGroup = (group: any) => {
     if (group.hasReadUntil) {
       return (
-        <div key="hasReadUntil" className="has-read-until">
+        <div key={group.hasReadUntil} className="has-read-until">
           <span>{shortDateTimeFormat(group.hasReadUntil)}</span>
         </div>
       )
     }
-
-    const key = group.messages[0].time
+    console.log('Group', group)
+    const key = `${group.messages[0].time}${group.messages[0].payload.message}`
```

Comment:
Doesn't the group's messages each have an `id` that can be used instead of `payload.message`?

Reply:
It doesn't seem to have a group id.

Reply:
the id of the first message in the group?